packages/main/src/Input.ts

```diff
@@ -647,7 +639,16 @@ class Input extends UI5Element implements SuggestionComponent, IFormInputElement
 
 		if (this.showSuggestions) {
 			this.enableSuggestions();
-			this.suggestionObjects = this.Suggestions!.defaultSlotProperties(this.typedInValue);
+
+			this.suggestionItems.forEach(item => {
+				if (item.hasAttribute("ui5-suggestion-item")) {
+					(item as SuggestionItem).markupText = this.typedInValue ? this.Suggestions?.hightlightInput(((item as SuggestionItem).text), this.typedInValue) : encodeXML((item as SuggestionItem).text || "");
```

Comment:
You can extract this code into a function, document it what it does (or name it accordingly) and call it for the nested items as well.

Reply:
+1 please extract this as it is quite difficult to read